src/systems/material.js

```diff
@@ -180,7 +182,13 @@ module.exports.System = registerSystem('material', {
     if (data.src.tagName) {
       // Since `data.src` can be an element, parse out the string if necessary for the hash.
       data = utils.extendDeep({}, data);
-      data.src = data.src.getAttribute('src');
+      data.src = data.src.src;
+      if (!data.src && data.src.tagName === 'VIDEO') {
+        // Video elements can have source elements as well
+        data.sources = Array.prototype.slice.call(
```

Comment:
I specifically did not do this because on some browsers, querySelectorAll results are not usable as an Array, but need to be converted with Array.prototype.slice.call first.